fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,272 @@
+    error_bars : str or list of str
+        The name of the error metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+        Error bars quantify the bounds relative to the base metric.
+
+        Example:
+            If the error bar for a certain column is :code:`[0.01, 0.02]`
+            and the metric is 0.6,
+            then the plotted bounds will be :code:`[0.59, 0.62]`
+
+        Note:
+            The return of the error bar function should be a tuple of the lower
+            and upper errors. i.e. :code:`[0.01, 0.02]`
+
+    conf_intervals : str or list of str
+        The name of the confidence intervals to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+        Note:
+            The return of the error bar function should be a tuple of the lower
+            and upper bounds. i.e. :code:`[0.59, 0.62]`
+
+    subplots : bool, default=True
+        Whether or not to plot metrics on separate subplots
+
+    plot_error_labels : bool, default=False
+        Whether or not to plot numerical labels for the error bounds
+
+    text_precision_digits : int, default=4
+        The number of digits of precision to show for error labels
+
+    text_font_size : int, default=8
+        The font size to use for error labels
+
+    text_color : str, default="black"
+        The font color to use for error labels
+
+    text_ha : str, default="center"
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    _check_for_ambiguous_error_metric(error_bars, conf_intervals)
+    _check_for_valid_metrics_format(metrics)
+
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    error_bars = [error_bars] if isinstance(error_bars, str) else error_bars
+
+    df = metric_frame.by_group
+
+    # only plot metrics that aren't tuples (filters out metric errors)
+    if metrics is None:
+        metrics = []
+        for metric in list(df):
+            if not isinstance(df[metric][0], tuple):
+                metrics.append(metric)
+
+    _check_if_metrics_and_error_metrics_same_length(metrics, error_bars, conf_intervals)
+    _check_if_metrics_length_zero(metrics)
+
+    # plotting without error bars or confidence intervals
+    # Note: Returns early
+    if error_bars is None and conf_intervals is None:
+        axs = _plot_df(df, metrics, kind, subplots, **kwargs)
+
+        return axs
+
+    df_all_errors = pd.DataFrame([])
+    df_all_bounds = pd.DataFrame([])
+    # plotting with confidence intervals:
+    if conf_intervals is not None:
+        for metric, conf_interval in zip(metrics, conf_intervals):
+            _check_valid_conf_interval(df[conf_interval])
+            df_all_errors[metric] = abs(df[metric] - df[conf_interval])
```

Comment:
It is technically possibly for the metric value to be outside of its confidence interval, so this might need tweaking. (I'm not sure whether `yerr` in works with negative values.)

Reply:
Fixed this (with this example of just adding 1 to the lower and upper bounds)
![image](https://user-images.githubusercontent.com/25800515/129623602-7617f479-a939-4fad-ba74-59b8758282b7.png)